tests/check/check.go

```diff
@@ -196,3 +199,430 @@ func runTest(t *testing.T, test individualTest, params testParams, contextTupleT
 		}
 	})
 }
+
+var matrix = individualTest{
+	Name: "complete_testing_model",
+	Stages: []*stage{
+		{
+			Name: "save model",
+			Model: `
+model
+  schema 1.1
+type user
+type employee
+# yes, wildcard is a userset instead of a direct, makes it easier and tests the userset path
+type directs-user
+  relations
+    define direct: [user]
+    define direct_cond: [user with xcond]
+    define direct_wild: [user:*]
+    define direct_wild_cond: [user:* with xcond]
+    define direct_and_direct_cond: [user, user with xcond, employee]
+    define direct_and_direct_wild: [user, user:*, employee:*]
+    define direct_and_direct_wild_cond: [user, user:* with xcond]
+    define direct_cond_and_direct_wild: [user with xcond, user:*]
+    define direct_cond_and_direct_wild_cond: [user with xcond, user:* with xcond]
+    define direct_wildcard_and_direct_wildcard_cond: [user:*, user:* with xcond]
+    define computed: direct
+    define computed_cond: direct_cond
+    define computed_wild: direct_wild
+    define computed_wild_cond: direct_wild_cond
+    define computed_computed: computed
+    define computed_computed_computed: computed_computed
+    define or_computed: computed or computed_cond or direct_wild
+    define and_computed: computed_cond and computed_wild
+    define butnot_computed: computed_wild_cond but not computed_computed
```

Comment:
Do we also want to have union/intersection/exclusion that does not involve cond/wildcard?  What about mixing union/intersection/exclusion?

Reply:
Check `complexity3` and `complexity4`